test/cases/api.sh

```diff
@@ -302,7 +325,14 @@ function createReqFileAWS() {
   "customizations": {
     "packages": [
       "postgresql"
-    ]
+    ],
+    "subscription": {
```

Comment:
subscription must be enabled only on RHEL runners I think. For Fedora and CentOS there's nothing to subscribe to.

Reply:
I am surprised that nothing actually failed in this case. CC @Gundersanne

Reply:
> > Hi @atodorov ,
> > I've update the script, removed the "dnf install dos2unix" step, because in some branches(e.g. https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1350313301 ) it cannot get available repo. Perhaps because this RHEL version has not been released yet and no published repo for that.
> > ```
> > + ssh -oStrictHostKeyChecking=no -i /tmp/tmp.iCaR9Uweoz/id_azure cloud-user@20.86.164.130 sudo dnf -y install dos2unix
> > Updating Subscription Management repositories.
> > Failed to set locale, defaulting to C.UTF-8
> > Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".
> > ```
> 
> That's running on a rhel-8-x86_64 runner which is a released distro, 8.3 at this time. Something went wrong during subscription/repo enablement. Not sure what.

Hi @atodorov , do you think we still need this "package installation check" step? If not, we can just skip this issue. Thanks!

Reply:
Thanks! Updated. I decoupled the subscription block from the whole json and make it optional. I know it's not good to read but I have no better idea...